tools/topology/topology2/include/components/kpb.conf

```diff
@@ -0,0 +1,89 @@
+#
+# Common widget kpb
+#
+# A generic kpb widget. All attributes defined herein are namespaced
+# by alsatplg to "Object.Widget.kpb.N.attribute_name"
+#
+# Usage: this component can be used by instantiating it in the parent object. i.e.
+#
+# 	Object.Widget.kpb."N" {
+#
+#	}
+#
+# Where N is the unique instance number for the kpb object within the same alsaconf node.
+
+Class.Widget."kpb" {
+	#
+	# Pipeline ID for the kpb object
+	#
+	DefineAttribute."index" {}
+
+	#
+	# Kpb object instance
+	#
+	DefineAttribute."instance" {}
+
+	DefineAttribute."cpc" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	DefineAttribute."num_audio_formats" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	#include common component definition
+	<include/components/widget-common.conf>
+
+	#
+	# kpb component UUID
+	#
+	DefineAttribute."uuid" {
+		type "string"
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.uuid"
+	}
+
+	attributes {
+		#
+		# The kpb widget name would be constructed using the index and
+		# instance attributes. For example: "kpb.0.1".
+		#
+		!constructor [
+			"index"
+			"instance"
+		]
+
+		#
+		# mandatory attributes that must be provided when the class is instantiated
+		#
+		!mandatory [
+			"no_pm"
+			"uuid"
+		]
+
+		#
+		# immutable attributes cannot be modified in the object instance
+		#
+		!immutable [
+			"uuid"
+		]
+
+		unique	"instance"
+	}
+
+	#
+	# Default attributes for kpb
+	#
+	type		"effect"
+
+	num_audio_formats 1
+	#UUID: D8218443-5FF3-4A4C-B388-6CFE07B9562E
+	uuid		"43:84:21:d8:f3:5f:4c:4a:b3:88:6c:fe:07:b9:56:2e"
+	no_pm 		"true"
+	core_id	0
+	cpc 360000
+	num_sink_pins		1
+	num_source_pins		3
```

Comment:
are these correct? Isnt it the other way around? In the topology graph in the PR description you have 2 sinks and 1 source for the KPB

Reply:
I checked cavs FW, and it's 3. Let me check SOF FW. Thanks for pointing it out.